x-pack/legacy/plugins/ml/public/data_frame/pages/data_frame_new_pivot/components/step_define/switch_modal.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { FC } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiConfirmModal, EuiOverlayMask } from '@elastic/eui';
+
+interface Props {
+  onCancel: () => void;
+  onConfirm: () => void;
+  type: 'pivot' | 'source';
```

Comment:
nit: This could be an `enum` so also later on in the component we wouldn't rely on plain strings.